src/network/message_blockdata.rs

```diff
@@ -43,6 +43,20 @@ pub enum Inventory {
     WitnessBlock(BlockHash),
 }
 
+impl Inventory {
+    /// Return the item value represented as a SHA256-d hash.
+    pub fn hash(&self) -> sha256d::Hash {
```

Comment:
Not sure this is a good name for the function taking into account collision with non-bitcoin `Hash` trait (non-critical, but may be still important). It could be that `From<Inventory> for sha256d::Hash` will work better; or if we'd like to support optional, a `try_as_hash()` as a function name